kin-recovery/recovery-sample/src/main/java/recovery/sample/BackupAndRestoreActivity.java

```diff
@@ -1,13 +1,138 @@
 package recovery.sample;
 
+import android.content.Intent;
 import android.os.Bundle;
 import android.support.v7.app.AppCompatActivity;
+import android.view.View;
+import android.widget.Button;
+import android.widget.TextView;
+import android.widget.Toast;
+import kin.recovery.BackupAndRestoreManager;
+import kin.sdk.Balance;
+import kin.sdk.Environment;
+import kin.sdk.KinClient;
+import recovery.sample.BackupAndRestorePresenterImpl.NetWorkType;
 
-public class BackupAndRestoreActivity extends AppCompatActivity {
+public class BackupAndRestoreActivity extends AppCompatActivity implements IBackupAndRestoreView, View.OnClickListener {
+
+	private static final String TAG = BackupAndRestoreActivity.class.getSimpleName();
+
+	private IBackupAndRestorePresenter backupAndRestorePresenter;
+
+	private Button createNewAccount;
+	private Button backupCurrentAccount;
+	private Button recoverAccount;
+	private TextView restoredAccountBalance;
+	private TextView restoredAccountPublicAddress;
+	private View balanceProgress;
 
 	@Override
 	protected void onCreate(Bundle savedInstanceState) {
 		super.onCreate(savedInstanceState);
+		setContentView(R.layout.activity_backup_and_restore);
+		createNewAccount = findViewById(R.id.create_new_account);
+		backupCurrentAccount = findViewById(R.id.backup_current_account);
+		recoverAccount = findViewById(R.id.recover_account);
+		restoredAccountBalance = findViewById(R.id.balance_value);
+		restoredAccountPublicAddress = findViewById(R.id.public_address_value);
+//		balanceProgress = findViewById(R.id.balance_progress);
+
+		createNewAccount.setOnClickListener(this);
+		backupCurrentAccount.setOnClickListener(this);
+		recoverAccount.setOnClickListener(this);
+
+		// TODO: 13/03/2019 should we support production?
```

Comment:
we do it on sdk sample app, any reason not doing it here?

Reply:
Because what they will gain from production here? it is just a recovery, no?

Reply:
mmm, yes I agree, in the kin-sdk-sample app it makes more sense to have production